Algorithm.Framework/Alphas/EmaCrossAlphaModel.py

```diff
@@ -63,53 +63,49 @@ def OnSecuritiesChanged(self, algorithm, changes):
         Args:
             algorithm: The algorithm instance that experienced the change in securities
             changes: The security additions and removals from the algorithm'''
-        for added in changes.AddedSecurities:
-            symbolData = self.symbolDataBySymbol.get(added.Symbol)
-            if symbolData is None:
-                symbolData = SymbolData(added, self.fastPeriod, self.slowPeriod, algorithm, self.resolution)
-                self.symbolDataBySymbol[added.Symbol] = symbolData
-            else:
-                # a security that was already initialized was re-added, reset the indicators
-                symbolData.Fast.Reset()
-                symbolData.Slow.Reset()
-
-        for removed in changes.RemovedSecurities:
-            data = self.symbolDataBySymbol.pop(removed.Symbol, None)
-            if data is not None:
-                # clean up our consolidators
-                data.RemoveConsolidators()
+        added_symbols = []
+        for security in changes.AddedSecurities:
+            symbol = security.Symbol 
+            if symbol not in self.symbolDataBySymbol:
+                self.symbolDataBySymbol[symbol] = SymbolData(symbol, self.fastPeriod, self.slowPeriod, algorithm, self.resolution)
+                added_symbols.append(symbol)
+            
+        if added_symbols:
+            history = algorithm.History[TradeBar](added_symbols, self.slowPeriod, self.resolution)
```

Comment:
Using `TradeBar` explicitly wont work well with all asset types like for example forex. Ideally we should use `Consolidator.InputType`, think we should have a base class/helper method to handle this nicely for C# & python in a standard way

Reply:
I think we are waiting on https://github.com/QuantConnect/Lean/issues/7095 so that we can just request a `Slice` object instead of `TradeBar`/`QuoteBar`